src/components/shared/Numeral/Numeral.tsx

```diff
@@ -44,19 +45,21 @@ const Numeral = ({
   suffix,
   className,
   appearance,
+  trimSpaceBeforeSuffix,
   ...rest
 }: Props) => {
   const convertedValue = convertToDecimal(value, decimals || 0);
 
   const formattedValue = getFormattedNumeralValue(convertedValue, value);
+
   return (
     <span
       className={classNames(getMainClasses(appearance, styles), className)}
       {...rest}
     >
       {prefix && `${prefix} `}
       {formattedValue}
-      {suffix && ` ${suffix}`}
+      {suffix && (trimSpaceBeforeSuffix ? suffix : ` ${suffix}`)}
```

Comment:
while this will work, it could be nicer to refactor all suffixes that need a leading whitespace to have it and just leave this component intact